tests/integration/test_restore_external_engines/test.py

```diff
@@ -0,0 +1,143 @@
+
+def fill_nodes(nodes, dbname):
+    cleanup_nodes(nodes, dbname)
+    for node in nodes:
+        node.query(f"CREATE DATABASE {dbname} ENGINE = Replicated('/clickhouse/databases/{dbname}', 'default', '{node.name}')")
+
+def drop_mysql_table(conn, tableName):
+    with conn.cursor() as cursor:
+        cursor.execute(f"DROP TABLE IF EXISTS `clickhouse`.`{tableName}`")
+
+def get_mysql_conn(cluster):
+    conn = pymysql.connect(
+        user="root", password="clickhouse", host=cluster.mysql8_ip, port=cluster.mysql8_port
+    )
+    return conn
+
+def fill_tables(cluster, dbname):
+    fill_nodes(nodes, dbname)
+
+    conn = get_mysql_conn(cluster)
+
+    with conn.cursor() as cursor:
+        cursor.execute(
+            "DROP DATABASE IF EXISTS clickhouse"
+        )
+        cursor.execute(
+            "CREATE DATABASE clickhouse"
+        )
+        cursor.execute(
+            "DROP TABLE IF EXISTS clickhouse.inference_table"
+        )
+        cursor.execute(
+            "CREATE TABLE clickhouse.inference_table (id INT PRIMARY KEY, data BINARY(16) NOT NULL)"
+        )
+        cursor.execute(
+            "INSERT INTO clickhouse.inference_table VALUES (100, X'9fad5e9eefdfb449')"
+        )
+        conn.commit()
+
+    parameters = "'mysql80:3306', 'clickhouse', 'inference_table', 'root', 'clickhouse'"
+
+    node1.query(
+        f"CREATE TABLE {dbname}.mysql_schema_inference_engine ENGINE=MySQL({parameters})"
+    )
+    node1.query(f"CREATE TABLE {dbname}.mysql_schema_inference_function AS mysql({parameters})")
+
+    node1.query(f"CREATE TABLE {dbname}.merge_tree (id UInt64, b String) ORDER BY id")
+    node1.query(f"INSERT INTO {dbname}.merge_tree VALUES (100, 'abc')")
+
+    expected = "id\tInt32\t\t\t\t\t\ndata\tFixedString(16)\t\t\t\t\t\n"
+    assert node1.query(f"DESCRIBE TABLE {dbname}.mysql_schema_inference_engine") == expected
+    assert node1.query(f"DESCRIBE TABLE {dbname}.mysql_schema_inference_function") == expected
+    assert node1.query(f"SELECT id FROM mysql({parameters})") == "100\n"
+    assert node1.query(f"SELECT id FROM {dbname}.mysql_schema_inference_engine") == "100\n"
+    assert node1.query(f"SELECT id FROM {dbname}.mysql_schema_inference_function") == "100\n"
+    assert node1.query(f"SELECT id FROM {dbname}.merge_tree") == "100\n"
+
+
+@pytest.fixture(scope="module")
+def start_cluster():
+    try:
+        cluster.start()
+        yield cluster
+
+    except Exception as ex:
+        print(ex)
+
+    finally:
+        cluster.shutdown()
+
+def test_restore_table(start_cluster):
+    fill_tables(cluster, "replicated")
+    backup_name = new_backup_name()
+    node2.query(f"SYSTEM SYNC DATABASE REPLICA replicated;")
+
+    node2.query(f"BACKUP DATABASE replicated TO {backup_name}")
+
+    node2.query("DROP TABLE replicated.mysql_schema_inference_engine")
+    node2.query("DROP TABLE replicated.mysql_schema_inference_function")
+
+    node3.query(f"SYSTEM SYNC DATABASE REPLICA replicated;")
```

Comment:
Unless there's a reason I'm not aware of, I find misleading that some queries end with `;` while others do not. I think we can be more consistent in that regard.